syntax = "proto3";

package voice;

// Voice Service for TTS (text-to-speech) and STT (speech-to-text)
service VoiceService {
  // Generate speech from text, streaming audio chunks
  rpc GenerateSpeech(GenerateSpeechRequest) returns (stream AudioChunk);
  
  // Transcribe audio to text
  rpc TranscribeAudio(TranscribeAudioRequest) returns (TranscribeAudioResponse);
  
  // Health check endpoint
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

// Request to generate speech
message GenerateSpeechRequest {
  string text = 1;
  optional string voice = 2;
}

// Audio chunk response
message AudioChunk {
  bytes audio_data = 1;
  bool is_final = 2;
}

// Request to transcribe audio
message TranscribeAudioRequest {
  bytes audio_data = 1;  // PCM audio data (16-bit, 16kHz, mono)
  int32 sample_rate = 2; // Sample rate (default: 16000)
}

// Response from transcription
message TranscribeAudioResponse {
  string text = 1;  // Transcribed text
  bool is_final = 2; // Whether transcription is final
}

// Health check request
message HealthCheckRequest {
}

// Health check response
message HealthCheckResponse {
  bool healthy = 1;
  string message = 2;
}
